src/main/java/org/springframework/data/elasticsearch/core/DefaultResultMapper.java

```diff
@@ -158,12 +159,18 @@ private String buildJSONFromFields(Collection<SearchHitField> values) {
 			// Only deal with String because ES generated Ids are strings !
 			if (idProperty != null && idProperty.getType().isAssignableFrom(String.class)) {
 				Method setter = idProperty.getSetter();
-				if (setter != null) {
-					try {
+				try {
+					if (setter != null) {
 						setter.invoke(result, id);
-					} catch (Throwable t) {
-						t.printStackTrace();
+					} else if(idProperty.getField() != null){
+						final Field field = idProperty.getField();
+						if(!field.isAccessible()) {
+							field.setAccessible(true);
+						}
+						field.set(result, id);
 					}
+				} catch (Throwable t) {
+					t.printStackTrace();
```

Comment:
This is what I said :-)
I am not a member of the team. I'm just a user like you.
Since you make changes in this area of the code I suggest to follow the boy scout rule and improve it!